charts/openstack-cluster/DEBUGGING.md

```diff
@@ -0,0 +1,172 @@
+# Debugging
+
+## Cluster Creation Summary
+
+The following diagram represents a high-level summary of cluster creation process. Each stage is described in more detail below.
+
+```mermaid
+sequenceDiagram
+  actor U as User
+  participant M as Management Cluster
+  participant O as OpenStack Project
+  U ->> M: `helm install ...`
+  M ->> O: Create network resources
+  activate O
+  Note right of O: Resources may include network, subnet, router, <br/> security groups and a load balancer.
+  O -->> M: Wait for resource creation
+  deactivate O
+  M ->> O: Create first control plane node
+  create participant W as Workload Cluster
+  O ->> W: Create first server
+  W -->> M: Wait for workload cluster API server to be online
+  M ->> W: Install addons (helmreleases & manifests)
+  activate W
+  W -->> M: Wait for core addons (e.g. CNI) to be ready
+  M ->> O: Create remaining control plane and worker nodes
+  O ->> W: Create remaining servers
+  W -->> M: Wait for all addons to be ready
+  deactivate W
+  M -) U: Cluster ready
+```
```

Comment:
I'm still on the fence about whether this diagram is really useful here, happy to remove it depending on others' opinions.